bendigobank.rb

```diff
@@ -0,0 +1,87 @@
+require 'watir'
+require 'json'
+require 'nokogiri'
+require 'date'
+require_relative 'account'
+require_relative 'transaction'
+
+class Bendigobank
+
+  def self.execute
+    connect
+    fetch_accounts(connect)
+    fetch_transactions(connect)
+  end
+
+  def self.connect
+    Watir.logger.ignore :deprecations
+    browser = Watir::Browser.new :firefox
+    browser.goto 'http://demo.bendigobank.com.au'
+    browser.button(name: 'customer_type').click
+    return browser
+  end
+
+  def self.fetch_accounts(browser)
+    html = Nokogiri::HTML.fragment(browser.div(class: "_23LTBXgogQ").html)
+    parse_accounts(html)
+  end
+
+  def self.fetch_transactions(accounts)
+    date = Time.new().to_datetime
+    datepast = date << 2
+    if datepast.month < 10
+      datepast = datepast.day.to_s + "/" + "0" + datepast.month.to_s + "/" + datepast.year.to_s
+    end
+    if date.month < 10 && date.day < 10
+      date = "0" + date.day.to_s + "/" + "0" + date.month.to_s + "/" + date.year.to_s
+      datepast = "0" + datepast.day.to_s + "/" + datepast.month.to_s + "/" + datepast.year.to_s
+    elsif date.month < 10
+      date = date.day.to_s + "/" + "0" + date.month.to_s + "/" + date.year.to_s
+    elsif date.day < 10
+      date = "0" + date.day.to_s + "/" + date.month.to_s + "/" + date.year.to_s
+      datepast = "0" + datepast.day.to_s + "/" + datepast.month.to_s + "/" + datepast.year.to_s
+    end
+     accounts.ol(class: "grouped-list__group__items").each do |li|
+       li.a(class: 'g9Ab3g8sIZ').click
+       accounts.i(class: 'ico-nav-bar-filter_16px').click
+       accounts.a(class: 'panel--bordered__item').click
+       accounts.ul(class: 'radio-group').li(index: 8).click
+       accounts.text_field(name: 'toDate').set(date)
+       accounts.text_field(name: 'fromDate').set(datepast)
+       accounts.button(class: 'button--primary').click
+       accounts.button(class: 'button--primary').click
+       html = Nokogiri::HTML.fragment(accounts.div(class: "activity-container").html)
+       transaction = Array[]
+       parse_transactions(html,transaction)
+     end
+  end
+
+  def self.parse_accounts(html)
+     html.css("ol.grouped-list__group__items li").each do |li|
+      name = li.css('._3jAwGcZ7sr').text
```

Comment:
Do not use hardcode value.

Reply:
You can set `binding.pry` at this code line and test this selector.

```
binding.pry
name = li at_css('span.some.correct.selector').text
```

Reply:
`.css` takes an array of Nokogiri elements   
`at_css` takes the first element of an array.